packages/code-studio/src/settings/ThemeSectionContent.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useCallback } from 'react';
+import {
+  Item,
+  ItemKey,
+  Picker,
+  ThemePicker,
+  useTheme,
+} from '@deephaven/components';
+import { assertNotNull } from '@deephaven/utils';
+import { useDispatch, useSelector } from 'react-redux';
+import { getSettings, updateSettings, type RootState } from '@deephaven/redux';
+
+export function ThemeSectionContent(): JSX.Element {
+  const theme = useTheme();
+  const settings = useSelector<RootState, ReturnType<typeof getSettings>>(
+    getSettings
+  );
+  const dispatch = useDispatch();
+
+  const updateDensity = useCallback(
+    (density: ItemKey | null) => {
+      if (
+        density !== 'normal' &&
+        density !== 'compact' &&
+        density !== 'spacious'
+      ) {
```

Comment:
If we had these values defined as a constant that we could do a `DENSITY_OPTIONS.includes(density)` instead, that would be ideal. This isn't typed at all right now.
When declaring `gridDensity` type, could do:
```
const DENSITY_OPTIONS = ['compact', 'regular', 'spacious'] as const
type Density = typeof DENSITY_OPTIONS[number]
```

Reply:
True. Since it was only 3 values and only in 1 place that I needed to do this (because of Picker saying it can be `string | int`) I was indifferent on doing something like the array. Although I guess I could create the items based on the array too

Reply:
Actually the types get more annoying than just doing this check IMO. `Array.includes` types based on the array type, so if you use a const string literal array, it only accepts values that are known to be in the array (which is pretty dumb IMO). The reasoning I saw in a closed issue was "you could be checking for a number in an array of strings which is probably wrong".

I'm just going to leave this as is I think